### ISSUE TITLE
`pkgx list-installed`

### ISSUE THREAD

**Issue by felipecrs** [pkgxdev/pkgx#985]
I'm aiming to [fully replace Homebrew with pkgx in my dotfiles](https://github.com/felipecrs/dotfiles/issues/101).

However, my dotfiles:

- Automatically installs missing tools
- Automatically uninstalls some tools I no longer want

And in order to perform that with pkgx, I need to fetch a list of pkgs already installed. Something like:

```bash
$ pkgx list-installed
openjdk.org@11
nodejs.org
yarnpkg.com

# a tool I want to have installed
$ if ! pkgx list-installed | grep -w openjdk.org@11; then pkgx install openjdk.org@11; fi

# a tool I do not want to have
$ if pkgx list-installed | grep -w asciinema.org; then pkgx uninstall asciinema.org; fi
```

**Comment by jhheider** [comment]
That seems like something that could (and probably should) easily be added to https://github.com/pkgxdev/mash. In its simplest form, it's really something like `grep -l pkgx ~/.local/bin/*`.

**Comment by felipecrs** [comment]
I ended up with:

```console
grep -rIhoP 'exec pkgx \+\K[^ ]+' "${HOME}/.local/bin" | sort -u
```

Or the uncompressed variant:

```console
grep --recursive --binary-files=without-match --no-filename --only-matching --perl-regexp 'exec pkgx \+\K[^ ]+' "${HOME}/.local/bin" | sort --unique
```

**Comment by felipecrs** [comment]
Here's an one-liner to uninstall all existing pkgx installed apps:

```console
grep -rIhoP 'exec pkgx \+\K[^ ]+' "${HOME}/.local/bin" | sort -u | xargs -rn1 --verbose pkgx uninstall
```

**Comment by felipecrs** [comment]
And just in case others may find it useful, I built logic in my dotfiles to automate management of pkgx installed stuff:

1. Check if a pkgx package is installed or not ([here](https://github.com/felipecrs/dotfiles/blob/5e6f7094814d4a5ef8a2cb8ae901169b203517af/home/.chezmoiscripts/run_after_30-install-pkgx-packages.sh.tmpl#L10-L15))

2. Refresh/recreate/update all stubs ([here](https://github.com/felipecrs/dotfiles/blob/5e6f7094814d4a5ef8a2cb8ae901169b203517af/home/dot_local/bin/executable_full-upgrade.tmpl#L47-L73))

3. Updating packages for installed stubs without installing them in case they have not been executed yet ([here](https://github.com/felipecrs/dotfiles/blob/5e6f7094814d4a5ef8a2cb8ae901169b203517af/home/dot_local/bin/executable_full-upgrade.tmpl#L75-L88))

**Comment by mxcl** [comment]
Yeah we're leaning towards mash scripts for now here. I think for pkgx v2 the install command will become its own package built on `pkgx` tech, as we intend for `dev`.

**Comment by iyefrat** [comment]
I think that having something like this (also as a mash script) would be a significant usability improvement. There is already https://github.com/pkgxdev/mash/blob/main/scripts/pantry-inventory.ts, a few more similar scripts that are documented on the website would make the project much easier to use as a brew replacement.

**Comment by mxcl** [comment]
```sh
pkgx mash pkgx/ls
```

We can expand this as you need. Can be as fully featured as desired since it is a totally separate thing.

**Comment by felipecrs** [comment]
Shouldn't I be looking for `pkgm ls` instead?

**Comment by mxcl** [comment]
oh indeed `pkgm ls` is what you want, though it doesn't actually work yet, it's pretty trivial to add.
